src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -1,244 +1,20 @@
-        outFile.deleteOnExit();
-        final File originalFile = new File(TEST_DATA_DIR, fileIn);
-        final File refFile;
-        final String[] args;
-        if (reference == null) {
-            refFile = null;
-            args = new String[]{
-                    "--input", originalFile.getAbsolutePath(),
-                    "--output", outFile.getAbsolutePath(),
-            };
-        } else {
-            refFile = new File(TEST_DATA_DIR, reference);
-            args = new String[]{
-                    "--input", originalFile.getAbsolutePath(),
-                    "--output", outFile.getAbsolutePath(),
-                    "-R", refFile.getAbsolutePath()
-            };
-        }
-        try (ReadsDataSource ds = reference==null ? new ReadsDataSource(originalFile.toPath()) :
-                new ReadsDataSource(originalFile.toPath(), SamReaderFactory.make().referenceSequence(refFile.toPath()))){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.queryname);
-        }
-
-        runCommandLine(args);
-        SamAssertionUtils.assertSamsEqual(outFile, originalFile, refFile);
-    }
-
-    @Test( groups = "spark")
-    public void testNameSorted() throws Exception {
-        final File inBam = new File(getTestDataDir(), "print_reads.bam");
-        try (ReadsDataSource ds = new ReadsDataSource(inBam.toPath())){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.queryname);
-        }
-        final File outBam = GATKBaseTest.createTempFile("print_reads", ".bam");
-        ArgumentsBuilder args = new ArgumentsBuilder();
-        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
-        args.add(inBam.getCanonicalPath());
-        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
-        args.add(outBam.getCanonicalPath());
-
-        this.runCommandLine(args.getArgsArray());
-
-        SamAssertionUtils.assertSamsEqual(outBam, inBam);
-    }
-
-    @Test( groups = "spark")
-    public void testUnSorted() throws Exception {
-        final File inBam = new File(getTestDataDir(), "print_reads.unsorted.bam");
-        try (ReadsDataSource ds = new ReadsDataSource(inBam.toPath())){
-            Assert.assertEquals(ds.getHeader().getSortOrder(), SAMFileHeader.SortOrder.unsorted);
-        }
-        final File outBam = GATKBaseTest.createTempFile("print_reads", ".bam");
-        ArgumentsBuilder args = new ArgumentsBuilder();
-        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
-        args.add(inBam.getCanonicalPath());
-        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
-        args.add(outBam.getCanonicalPath());
-
-        this.runCommandLine(args.getArgsArray());
+import java.io.File;
 
-        SamAssertionUtils.assertSamsEqual(outBam, inBam);
-    }
+public final class PrintReadsSparkIntegrationTest extends AbstractPrintReadsIntegrationTest {
```

Comment:
We still want these tests to be tagged as spark tests I think, i'm not sure how the propagation for testng works with inherited tests but these tests should probably stay in the spark group. If you added `@Test(groups = {"spark"})` to the class I suspect this might propagate the test tags correctly.

Reply:
You can check by looking at the travis test outputs to check that these did indeed run with the spark tests.

Reply:
I added the `spark` group back. Also, these tests were all run as a part of the `integration` test type.